### ISSUE TITLE
Button5 cannot be assigned to G5

### ISSUE THREAD

**Issue by adrfantini** [krayon/ratslap#9]
I have a very weird behaviour here. After having modified many buttons to suit my needs, I have the following problem.

I cannot assign many software mouse Buttons to many hardware buttons. Assigning keys seems to always work tho.

Say I am in this situation:
```
╰─[XPS15af]─>  sudo ratslap --p F3
RatSlap v0.2.1-1-gd6b9714 (BUILT: 2017-04-07 16:39:56+0200)
Copyright (C) 2016 Todd Harbour
Linux configuration tool for Logitech mice (currently only G300/G300S)
https://gitlab.com/krayon/ratslap/
Found Logitech G300s (046d:c246) @ 0xfcba80
Detaching kernel driver...
Printing Mode: F3
  Colour:              cyan
  Report Rate:         1000
  DPI #1:               500
  DPI #2:        (DEF) 1000
  DPI #3:              1500
  DPI #4:              2500
  DPI Shift:           NOT SET
  Left Click (But1):   Button1
  Right Click (But2):  Button2
  Middle Click (But3): Button3
  G4:                  
  G5:                  
  G6:                  
  G7:                  
  G8:                  ModeSwitch
  G9:                  DPICycle
Attaching kernel driver...
```

Now let's try to modify it with:
```
╰─[XPS15af]─>  sudo ratslap --modify F3 --G4 Button4
RatSlap v0.2.1-1-gd6b9714 (BUILT: 2017-04-07 16:39:56+0200)
Copyright (C) 2016 Todd Harbour
Linux configuration tool for Logitech mice (currently only G300/G300S)
https://gitlab.com/krayon/ratslap/
Found Logitech G300s (046d:c246) @ 0x13aea80
Detaching kernel driver...
Modifying Mode: F3
    Setting button 4: Button4
Saving Mode: F3
Attaching kernel driver...
```
Result:
```
╰─[XPS15af]─>  sudo ratslap --p F3
RatSlap v0.2.1-1-gd6b9714 (BUILT: 2017-04-07 16:39:56+0200)
Copyright (C) 2016 Todd Harbour
Linux configuration tool for Logitech mice (currently only G300/G300S)
https://gitlab.com/krayon/ratslap/
Found Logitech G300s (046d:c246) @ 0x1a5ba80
Detaching kernel driver...
Printing Mode: F3
  Colour:              cyan
  Report Rate:         1000
  DPI #1:               500
  DPI #2:        (DEF) 1000
  DPI #3:              1500
  DPI #4:              2500
  DPI Shift:           NOT SET
  Left Click (But1):   Button1
  Right Click (But2):  Button2
  Middle Click (But3): Button3
  G4:                  
  G5:                  
  G6:                  
  G7:                  
  G8:                  ModeSwitch
  G9:                  DPICycle
Attaching kernel driver...
```

Nothing! Let's try assigning keys instead:
```
sudo ratslap --modify F3 --G4 LeftAlt+K
RatSlap v0.2.1-1-gd6b9714 (BUILT: 2017-04-07 16:39:56+0200)
Copyright (C) 2016 Todd Harbour
Linux configuration tool for Logitech mice (currently only G300/G300S)
https://gitlab.com/krayon/ratslap/
Found Logitech G300s (046d:c246) @ 0xe9fa80
Detaching kernel driver...
Modifying Mode: F3
    Setting button 4: LeftAlt+K
Saving Mode: F3
Attaching kernel driver...
┌─[✓]─[21:51:59]─[+80%]─[39°C]─[pwd:~]
╰─[XPS15af]─>  sudo ratslap --p F3
RatSlap v0.2.1-1-gd6b9714 (BUILT: 2017-04-07 16:39:56+0200)
Copyright (C) 2016 Todd Harbour
Linux configuration tool for Logitech mice (currently only G300/G300S)
https://gitlab.com/krayon/ratslap/
Found Logitech G300s (046d:c246) @ 0x10efa80
Detaching kernel driver...
Printing Mode: F3
  Colour:              cyan
  Report Rate:         1000
  DPI #1:               500
  DPI #2:        (DEF) 1000
  DPI #3:              1500
  DPI #4:              2500
  DPI Shift:           NOT SET
  Left Click (But1):   Button1
  Right Click (But2):  Button2
  Middle Click (But3): Button3
  G4:                  LeftAlt + K
  G5:                  
  G6:                  
  G7:                  
  G8:                  ModeSwitch
  G9:                  DPICycle
Attaching kernel driver...
```
Works...

In short, I can seem to bind mouse Buttons anymore!
However, some, like `sudo ratslap --modify F3 --G7 Button9` work, while others, like `sudo ratslap --modify F3 --G7 Button5` do not.

**Comment by krayon** [comment]
TRACKED AS QB#118 : https://bugs.qdnx.org/issue/118

**Comment by krayon** [comment]
RELATED: QB#119 (Unrecognised keys/buttons fail silently) : https://bugs.qdnx.org/issue/119

**Comment by krayon** [comment]
@adrfantini Unfortunately this is due to a limitation in the hardware. Not only can you not assign the scroll wheel to anything, you also can't assign buttons to the scroll wheel. In mice, the scroll wheel actually generates mouse button's 4 and 5 - hence why you can't assign G4 to either of these buttons.

It is indeed a bug however that this is not reported to the user.

**Comment by krayon** [comment]
Fixed [QB#118](https://bugs.qdnx.org/issue/118) and [QB#119](https://bugs.qdnx.org/issue/119)
